pgx/src/io_intf.ml

```diff
@@ -14,6 +14,18 @@ module type S = sig
     | Inet of string * int
 
   val open_connection : sockaddr -> (in_channel * out_channel) t
+
+  type ssl_config
+
+  val upgrade_ssl
+    : [ `Not_supported
+      | `Supported of
+        ?ssl_config:ssl_config
+        -> in_channel
+        -> out_channel
+        -> (in_channel * out_channel) t
+      ]
+
```

Comment:
Now that the `async_ssl` dependency is mandatory, will there still be a scenario where attempting to setup a tls connection via conduit will fail because of missing support? I'm guessing with the current setup we might not need the `Not_supported` branch at all?

Reply:
We need the Not_supported branch because Pgx_lwt and Pgx_unix don't support TLS right now. Hopefully at some point we can get this supported for Pgx_lwt, but I doubt we'll ever support it for Pgx_unix.

I'm open to alternative ways of doing this though if there is one?